components/net/image_cache.rs

```diff
@@ -523,7 +677,13 @@ impl ImageCacheStore {
     fn handle_decoder(&mut self, msg: DecoderMsg) {
         let image = match msg.image {
             None => LoadResult::None,
-            Some(DecodedImage::Raster(raster_image)) => LoadResult::LoadedRasterImage(raster_image),
+            Some(DecodedImage::Raster(raster_image)) => {
+                if raster_image.id.is_none() {
+                    self.load_image_with_keycache(PendingKey::RasterImage((msg.key, raster_image)));
+                    return;
+                }
```

Comment:
This conditional seems unnecessary, since at this point, according to the [logic on `main`](https://github.com/servo/servo/blob/824755d868308ed1658db8f5dabe54959e55590a/components/net/image_cache.rs#L449), the `id` is always `None` and a webrender key is requested and set.

Reply:
Should there then be an assertion or just completely removed?